OutOfSchool/OutOfSchool.ExternalFileStore.Impl/NotificationImplementations/RedisStorageNotificationHandler.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Microsoft.Extensions.Logging;
+using OutOfSchool.ExternalFileStore.Models;
+using OutOfSchool.ExternalFileStore.NotificationInterfaces;
+using StackExchange.Redis;
+
+namespace OutOfSchool.ExternalFileStore.NotificationImplementations;
+public class RedisStorageNotificationHandler : IStorageNotificationHandler, IDisposable
+{
+    private readonly IConnectionMultiplexer _redisConnection;
+    private readonly ILogger<RedisStorageNotificationHandler> _logger;
+    private readonly Dictionary<string, ISubscriber> _subscribers = new Dictionary<string, ISubscriber>();
+    private readonly IProcessNotificationService _processNotificationService;
+
+
+    public RedisStorageNotificationHandler(
+            IConnectionMultiplexer redisConnection,
```

Comment:
Do not inject redis connection directly, we have a subproject for managing redis. You can add the publish/subscribe methods to one of the interfaces tehre (and implemetnations)